### PR TITLE
[ Metrics ] Rewrite + Update Action Completion Concept Page

### PR DESCRIPTION
**Pull request by erinmikailstaples**
## Describe your changes

What did you change? Why did you change it?

## Shortcut ticket

For Galileo internally raised PRs only, please update this with your shortcut ticket.

[SC-39017](https://app.shortcut.com/galileo/story/39017/docs-individual-metric-page-action-completion)

## Checklist before requesting a review

- [ ] - Is this ready for review? If not, raise as a draft PR
- [ ] - This deployed to a staging environment correctly
- [ ] - I have reviewed my changes
- [ ] - I have reviewed the deployed version of my changes
- [ ] - I have tested any code that is added or updated
- [ ] - I have verified all images and videos are clear, with appropriate zoom
- [ ] - I have verified all images and videos match production (or dev for unreleased features)
- [ ] - I have tested that the content matches the functionality in production (or dev for unreleased features)
- [ ] - All checks have passed
- [ ] - This references a feature that is public. If not, add a note and we can schedule the merge for after the feature release

### CODE REVIEW

File: concepts/metrics/agentic/action-completion.mdx [hunk 1]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
 description: Understand how to measure whether your agent actually accomplished a user's goals
 ---
 
-import { Scale } from "/snippets/components/scale.mdx";
-import { DefinitionCard } from "/snippets/components/definition-card.mdx";
+import { MetricWhenToUse } from "/snippets/components/metric-when-to-use.mdx";
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:6]:
```suggestion
import { DefinitionCard } from "/snippets/components/definition-card.mdx";
import { MetricWhenToUse } from "/snippets/components/metric-when-to-use.mdx";
```

Don't you need this back in? Not seeing this card on the deployed page

File: concepts/metrics/agentic/action-completion.mdx [hunk 2]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
-Additionally, the response must:
-- Be coherent and factually accurate.
-- Comprehensively address every aspect of the user's request.
-- Avoid contradicting tool outputs.
-- Summarize all relevant parts returned by tools.
+Action Completion addresses the common pain points of agent performance by measuring whether AI agents are actually helping users achieve their end goal rather than just providing responses.
+
+
+Action Completion is successful when:
+- The agent provides a complete answer in the case of a question
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:18]: Is this list an AND or an OR?

File: concepts/metrics/agentic/action-completion.mdx [hunk 3]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
+- The response summarizes all relevant parts returned by tools
+
+### Action Completion at a glance
+
+| Property | Description |
+|:----------|:-------------|
+| **Name of Metric** | Action Completion |
+| **Metric Category** | Agentic Metrics |
+| **Use this metric for** | Measuring whether the agent successfully accomplished the user's goal |
+| **Can be applied to** | Agent, Session |
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:32]:
```suggestion
| **Can be applied to** | Agent span, Session |
```

Also what about trace?

File: concepts/metrics/agentic/action-completion.mdx [hunk 4]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
+| **LLM/Luna Support** | Supported with both LLM + Luna models |
+| **Protect Runtime Protection** | No |
+| **Constants** | None |
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:35]: What does constants mean?

File: concepts/metrics/agentic/action-completion.mdx [hunk 5]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
+| **Usage Context** | Agentic workflows, multi-step tasks, tool-using assistants |
+| **Value Type** | Percentage |
+| **Input/Output Requirements** | Requires agent responses and user goals for evaluation |
+
+### When to use this metric
+
+<MetricWhenToUse
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:42]: Maybe remove the H3 above, or edit this component to remove the title - looks weird having 2 when to use this metric headers

Image

Comment (jimbobbennett) [reply concepts/metrics/agentic/action-completion.mdx:42]: Thinking removing the H3 is better

File: concepts/metrics/agentic/action-completion.mdx [hunk 6]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
-Action completion will help determine whether the agent successfully accomplished all of the user's goals. By tracking Action Completion over time, teams can identify patterns where agents fall short, analyze why certain scenarios lead to failures, and focus on targeted fixes.
-</Card>
+| Issue | Cause | Solution |
+|:-------|:-------|:----------|
+| Incomplete responses | Agent stops before addressing all user requirements | Implement comprehensive response generation and ensure all user goals are explicitly addressed |
+| Tool output contradictions | Agent ignores or contradicts information from tools | Ensure agent properly summarizes and incorporates all relevant tool outputs without contradiction |
+| Missing confirmations | Agent doesn't confirm successful actions | Add explicit confirmation steps for action-based requests |
+| Factual inaccuracies | Agent provides incorrect information | Implement fact-checking mechanisms and ensure responses align with tool outputs |
+
+### Best practices for optimization:
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:125]: This section is repeated below. In action advancement this section has different tips

File: concepts/metrics/agentic/action-completion.mdx [hunk 7]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
-    When Action Completion scores are low, examine specific steps or scenarios where agents fail to meet user goals. Use this analysis to identify and address systematic issues.
-  </Card>
-  <Card title="Combine with Other Metrics" icon="scale-balanced">
-    Use Action Completion alongside other agentic metrics, such as Action Advancement, to get a comprehensive view of your assistant's effectiveness and identify areas for improvement.
-  </Card>
-  <Card title="Test Edge Cases" icon="code-branch">
-    Create evaluation datasets that include complex, multi-step tasks to thoroughly assess your agent's ability to handle challenging scenarios and advance user goals effectively.
-  </Card>
-</CardGroup>
+Implement Action Completion monitoring to ensure your agent consistently accomplishes user goals and provides complete, accurate responses.
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:146]: This feels duplicated from action advancement - should we remove this section from both and add to the overview page?

File: concepts/metrics/agentic/action-completion.mdx [hunk 8]
```diff
@@ -3,97 +3,168 @@ title: Action Completion
...
 <Note>
-When optimizing for Action Completion, ensure you're not sacrificing other important aspects like safety, factual accuracy, or user experience in pursuit of task completion.
+  When optimizing for Action Completion, ensure you're not sacrificing other important aspects like safety, factual accuracy, or user experience in pursuit of task completion.
 </Note>
+
+## Related Resources
+If you would like to dive deeper or start implementing Action Completion, check out the following resources:
+
+### How-to guides:
+- [Agentic AI Examples](/how-to-guides/agentic-ai/basic-example)
```
Comment (jimbobbennett) [inline concepts/metrics/agentic/action-completion.mdx:165]: Minor - can we change these for cards (also in action advancement and the template) just to make the CTAs stand out more

**Review by jimbobbennett — CHANGES_REQUESTED** [review]

File: concepts/metrics/agentic/action-completion.mdx (hunk 1 above)
Comment (erinmikailstaples) [reply concepts/metrics/agentic/action-completion.mdx:6]: ooh good catch — thanks for the set of eyes.

File: concepts/metrics/agentic/action-completion.mdx (hunk 2 above)
Comment (erinmikailstaples) [reply concepts/metrics/agentic/action-completion.mdx:18]: Shortcut: https://app.shortcut.com/galileo/story/39101/metrics-action-completion-how-do-we-know-these-metrics-are-correct

File: concepts/metrics/agentic/action-completion.mdx (hunk 8 above)
Comment (erinmikailstaples) [reply concepts/metrics/agentic/action-completion.mdx:165]: can do! will add this in a follow up PR!

**Comment by erinmikailstaples** [comment]
Related PRs incoming here: 
1. Card component added to template for CTA at the bottom of the page
2. Adjusting template based on that card component
3. Add "best practices" to the metric category page for each vs. on the individual metrics (create template)
4. Adjust Action Advancement to match these changes.

**Review by jimbobbennett — APPROVED** [review]
